pkg/ccl/spanconfigccl/spanconfigreconcilerccl/spanconfigreconcilerccl_test: TestDataDriven failed

pkg/ccl/spanconfigccl/spanconfigreconcilerccl/spanconfigreconcilerccl_test.TestDataDriven [failed](https://tanzanite.cluster.engflow.com/invocations/default/5a360898-f829-4a51-9285-d068ad6d9c1f?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3NwYW5jb25maWdjY2wvc3BhbmNvbmZpZ3JlY29uY2lsZXJjY2w6c3BhbmNvbmZpZ3JlY29uY2lsZXJjY2xfdGVzdA==) on release-24.1 @ [6e3e9e6012dc04dabc64333a87e77a80d9f9d46b](https://github.com/cockroachdb/cockroach/commits/6e3e9e6012dc04dabc64333a87e77a80d9f9d46b):


```
        /Tenant/10/Table/4{7-8}                    ignore_strict_gc=true rangefeed_enabled=true exclude_data_from_backup=true
        /Tenant/10/Table/4{8-9}                    database system (tenant)
        /Tenant/10/Table/5{0-1}                    database system (tenant)
        /Tenant/10/Table/5{1-2}                    database system (tenant)
        /Tenant/10/Table/5{2-3}                    database system (tenant)
        /Tenant/10/Table/5{3-4}                    database system (tenant)
        /Tenant/10/Table/5{4-5}                    database system (tenant)
        /Tenant/10/Table/5{5-6}                    database system (tenant)
        /Tenant/10/Table/5{6-7}                    database system (tenant)
        /Tenant/10/Table/5{7-8}                    database system (tenant)
        /Tenant/10/Table/5{8-9}                    database system (tenant)
        /Tenant/10/Table/{59-60}                   database system (tenant)
        /Tenant/10/Table/6{0-1}                    database system (tenant)
        /Tenant/10/Table/6{1-2}                    database system (tenant)
        /Tenant/10/Table/6{2-3}                    database system (tenant)
        /Tenant/10/Table/6{3-4}                    database system (tenant)
        /Tenant/10/Table/6{4-5}                    database system (tenant)
        /Tenant/10/Table/6{5-6}                    database system (tenant)
        /Tenant/10/Table/6{6-7}                    database system (tenant)
        /Tenant/11{-\x00}                          database system (tenant)
        /Tenant/12{-\x00}                          database system (tenant)
    datadriven.go:144: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/spanconfigreconcilerccl_test_/spanconfigreconcilerccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/testdata/multitenant/basic:195:
        exec-sql [1 args]
        CREATE DATABASE db;
        CREATE TABLE db.t1();
        CREATE TABLE db.t2();
        CREATE SCHEMA db.sc;
        CREATE TYPE typ AS ENUM();
        CREATE VIEW v AS SELECT 1;
        CREATE SEQUENCE db.seq;
        CREATE MATERIALIZED VIEW mv AS SELECT 1;
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/spanconfigreconcilerccl_test_/spanconfigreconcilerccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/testdata/multitenant/basic:206:
         
        expected:
        upsert /Tenant/10/Table/10{6-7}            range default
        upsert /Tenant/10/Table/10{7-8}            range default
        upsert /Tenant/10/Table/11{2-3}            range default
        upsert /Tenant/10/Table/11{3-4}            range default
        
        found:
        upsert /Tenant/10/Table/11{6-7}            range default
        upsert /Tenant/10/Table/11{7-8}            range default
        upsert /Tenant/10/Table/12{2-3}            range default
        upsert /Tenant/10/Table/12{3-4}            range default
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestDataDriven_multitenant_basic4110065205
        --- FAIL: TestDataDriven/multitenant/basic (24.81s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-37817

--- a/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/BUILD.bazel
+++ b/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/BUILD.bazel
@@ -21,6 +21,7 @@ go_test(
         "//pkg/spanconfig",
         "//pkg/spanconfig/spanconfigtestutils",
         "//pkg/spanconfig/spanconfigtestutils/spanconfigtestcluster",
+        "//pkg/sql",
         "//pkg/sql/sqlliveness/sqllivenesstestutils",
         "//pkg/testutils",
         "//pkg/testutils/datapathutils",

--- a/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/datadriven_test.go
+++ b/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/datadriven_test.go
@@ -23,6 +23,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/spanconfig"
 	"github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigtestutils"
 	"github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigtestutils/spanconfigtestcluster"
+	"github.com/cockroachdb/cockroach/pkg/sql"
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlliveness/sqllivenesstestutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/datapathutils"
@@ -106,6 +107,9 @@ func TestDataDriven(t *testing.T) {
 				Knobs: base.TestingKnobs{
 					JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals(), // speeds up test
 					SpanConfig:       scKnobs,
+					SQLExecutor: &sql.ExecutorTestingKnobs{
+						UseTransactionalDescIDGenerator: true,
+					},
 				},
 			},
 		})

